cypress/integration/tx_modal.spec.js

```diff
@@ -0,0 +1,203 @@
+    cy.contains('a', 'Accept selection').click()
+  })
+
+  describe('Send funds modal', () => {
+    describe('Send Funds form', () => {
+      before(() => {
+        // Open Send Funds Modal
+        cy.contains('New Transaction').click()
+        cy.contains('Send funds').click()
+      })
+
+      it('should display Send Funds modal with all the form elements', () => {
+        // Modal header
+        cy.get('form')
+          .prev()
+          .prev()
+          .within(() => {
+            // Send Funds modal is open
+            cy.contains('Send funds').should('be.visible')
+            cy.contains('Step 1 of 2').should('be.visible')
+
+            // Current network is same as Safe
+            cy.contains('Rinkeby').should('be.visible')
+          })
+
+        // It contains the form elements
+        cy.get('form').within(() => {
+          // Sending from the current Safe address
+          const [chainPrefix, safeAddress] = TEST_SAFE.split(':')
+          cy.contains(chainPrefix)
+          cy.contains(safeAddress)
+
+          // Current ETH balance
+          cy.contains('Balance:').contains('0.3 ETH')
+
+          // Recipient field
+          cy.get('#address-book-input').should('be.visible')
+
+          // Token selector
+          cy.contains('Select an asset*').should('be.visible')
+
+          // Amount field
+          cy.contains('Amount').should('be.visible')
+        })
+
+        // Review button is disabled
+        cy.get('button[type="submit"]').should('be.disabled')
+      })
+
+      it('should resolve the ENS name', () => {
+        // Fills recipient with ens
+        cy.get('label[for="address-book-input"]').next().type(RECIPIENT_ENS)
+
+        // Waits for resolving the ENS
+        cy.contains(RECIPIENT_ADDRESS).should('be.visible')
+      })
+
+      it('should have all tokens available in the token selector', () => {
+        // Click on the Token selector
+        cy.contains('Select an asset*').click()
+
+        const ownedTokens = ['Dai', 'Wrapped Ether', 'Ether', 'Uniswap', 'Gnosis', '0x', 'USD Coin']
+        ownedTokens.forEach((token) => {
+          cy.get('ul[role="listbox"]').contains(token)
+        })
+      })
+
+      it('should validate token amount', () => {
+        // Select a token
+        cy.get('ul[role="listbox"]').contains('Gnosis').click()
+
+        // Insert an incorrect amount
+        cy.get('input[placeholder="Amount*"]').type('0.4')
+
+        // Selecting more than the balance is not allowed
+        cy.contains('Maximum value is 0.000004')
+
+        // Form field contains an error class
+        cy.get('input[placeholder="Amount*"]')
+          // Parent div is MuiInputBase-root
+          .parent('div')
+          .should(($div) => {
+            // Turn the classList into an array
+            const classList = Array.from($div[0].classList)
+            expect(classList).to.include('MuiInputBase-root').and.to.include('Mui-error')
+          })
```

Comment:
This is also an implementation detail. It would be more robust to check by content e.g. for the existence of an error message below the input field.

Reply:
I must have deleted the error label assertion when moving things around. Nice catch!

Reply:
Oh wait! It's in the line above
>// Selecting more than the balance is not allowed
        cy.contains('Maximum value is 0.000004')
        
do you think we shouldn't have both?

Reply:
Let's keep both because of robustness 💪 and if we move away form MUI we remove the one checking the classes.
I like that it checks the field doesn't have the error class and then it finds it though I understand perfectly your argument.